File: Sources/BugNarrator/Utilities/BugNarratorLinks.swift
import Foundation

enum BugNarratorLinks {
    static let repository = URL(string: "https://github.com/ABD-Enterprises/bug-narrator")!
    static let documentation = URL(string: "https://github.com/ABD-Enterprises/bug-narrator/blob/main/docs/UserGuide.md")!
    static let issues = URL(string: "https://github.com/ABD-Enterprises/bug-narrator/issues/new")!
    static let releases = URL(string: "https://github.com/ABD-Enterprises/bug-narrator/releases")!
    static let supportDevelopment = URL(string: "https://www.paypal.com/donate/?hosted_button_id=FWFQ6KCZBWWH8")!
    static let generateGitHubToken = URL(string: "https://github.com/settings/tokens/new?scopes=repo&description=BugNarrator")!
    static let generateJiraToken = URL(string: "https://id.atlassian.com/manage-profile/security/api-tokens")!
    static let microphonePrivacySettings = URL(string: "x-apple.systempreferences:com.apple.preference.security?Privacy_Microphone")!
    static let screenRecordingPrivacySettings = URL(string: "x-apple.systempreferences:com.apple.preference.security?Privacy_ScreenCapture")!
    static let securityPrivacySettings = URL(fileURLWithPath: "/System/Library/PreferencePanes/Security.prefPane")
    static let systemSettingsApp = URL(fileURLWithPath: "/System/Applications/System Settings.app")
}

File: windows/src/BugNarrator.Windows.Services/Shell/BugNarratorLinks.cs
namespace BugNarrator.Windows.Services.Shell;

/// <summary>
/// The same destinations macOS declares in Sources/BugNarrator/Utilities/BugNarratorLinks.swift,
/// byte for byte, so the two platforms' menus lead to the same places. Tests pin each string.
/// </summary>
public static class BugNarratorLinks
{
    public const string Repository = "https://github.com/ABD-Enterprises/bug-narrator";
    public const string Documentation = "https://github.com/ABD-Enterprises/bug-narrator/blob/main/docs/UserGuide.md";
    public const string Issues = "https://github.com/ABD-Enterprises/bug-narrator/issues/new";
    public const string Releases = "https://github.com/ABD-Enterprises/bug-narrator/releases";
    public const string SupportDevelopment = "https://www.paypal.com/donate/?hosted_button_id=FWFQ6KCZBWWH8";
}

/// <summary>What activating a tray entry does; the tray raises the matching request.</summary>
public enum TrayMenuEntryKind
{
    Action,
    Url,
}

/// <summary>One tray menu entry as the presentation layer describes it, so tests can assert the menu without WinForms.</summary>
public sealed record TrayMenuEntry(string Label, TrayMenuEntryKind Kind, string? Url = null);
